server/controllers/items/recent_public.coffee

```diff
@@ -0,0 +1,40 @@
+__ = require('config').universalPath
+_ = __.require 'builders', 'utils'
+error_ = __.require 'lib', 'error/error'
+items_ = __.require 'controllers', 'items/lib/items'
+user_ = __.require 'controllers', 'user/lib/user'
+
+module.exports = (req, res)->
+  { query } = req
+  assertImage = query['assert-image'] is 'true'
+  reqUserId = req.user?._id
+
+  items_.publicByDate 100, 0, assertImage, reqUserId
+  .then selectRecentItems
+  .then bundleOwnersData.bind(null, res, reqUserId)
+  .catch error_.Handler(req, res)
+
+selectRecentItems = (items)->
+  itemsLimit = 15
+  maxItemsPerOwner = 3
+
+  Promise.resolve _.groupBy items, (item)-> item.owner
+  .then (itemsByOwner)->
+    _.map itemsByOwner, (items)->
+      items.slice 0, maxItemsPerOwner
+  .then (limitedItemsByOwner)->
+    _.flatten _.values limitedItemsByOwner
+  .then (allRecentItems)->
+    allRecentItems.slice 0, itemsLimit
```

Comment:
lines 21-28 could be refactored with [Lodash sequence methods](https://lodash.com/docs/#lodash) as
```js
  _(items)
  .groupBy _.property('owner')
  .map (items)-> items.slice(0, maxItemsPerOwner)
  .values()
  .flatten()
  .slice 0, itemsLimit
  .value()
```